ee/control/actionqueue/actionqueue.go

```diff
@@ -0,0 +1,241 @@
+package actionqueue
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/agent/storage/inmemory"
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+const (
+	ActionsSubsystem       = "actions"
+	defaultCleanupInterval = time.Hour * 12
+	// about 6 months, long enough to ensure that K2 no longer has the message
+	// and will not send a duplicate
+	actionRetentionPeriod = time.Hour * 24 * 30 * 6
+)
+
+type updater interface {
+	Update(data io.Reader) error
```

Comment:
I think the reuse of updater/`Update` is a little overloaded (since it's the same function as for consumers in the control system) and isn't as precise as it could be here. Action "updaters" aren't really updaters because they're not performing generic data updates -- they're taking actions. What about `actor interface` and `Action(data io.Reader)` or `TakeAction(data io.Reader)`? I understand if we feel like that might be overloading run group stuff, though -- maybe something with "execute" or "perform" instead?

Reply:
yea, I agree it's overloaded, went with `actor.Do`

Reply:
Ah that's way better/simpler than Action/TakeAction, I like it! Thanks for updating